### PR TITLE
simplify ui elements

### PR DESCRIPTION
**Pull request by YousefED**
Proposal for a simplified setup of the ui elements

### CODE REVIEW

File: packages/core/src/shared/plugins/suggestion/SuggestionPlugin.ts [hunk 1]
```diff
@@ -64,24 +40,6 @@ class SuggestionsMenuView<
       updateSuggestionsMenu(this.suggestionsMenuState);
     };
 
-    this.itemCallback = (item: T) => {
```
Comment (YousefED) [inline packages/core/src/shared/plugins/suggestion/SuggestionPlugin.ts:67]: I don't think this was used

File: packages/react/src/SlashMenu/components/SlashMenuPositioner.tsx [hunk 2]
```diff
@@ -36,21 +31,15 @@ export const SlashMenuPositioner = <
     useState<number>();
 
   const referencePos = useRef<DOMRect>();
-  const callbacks =
-    useRef<SuggestionsMenuCallbacks<ReactSlashMenuItem<BSchema>>>();
 
   useEffect(() => {
-    callbacks.current = props.editor.createSlashMenu((slashMenuState) => {
+    return props.editor.slashMenu.on("update", (slashMenuState) => {
```
Comment (matthewlipski) [inline packages/react/src/SlashMenu/components/SlashMenuPositioner.tsx:36]: Is there any point having the first param? Isn't it always just "update"? Same for the other UI elements

File: packages/core/src/shared/EventEmitter.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+type CallbackType<
+  T extends Record<string, any>,
+  EventName extends StringKeyOf<T>
+> = T[EventName] extends any[] ? T[EventName] : [T[EventName]];
+type CallbackFunction<
+  T extends Record<string, any>,
+  EventName extends StringKeyOf<T>
+> = (...props: CallbackType<T, EventName>) => any;
+
+export class EventEmitter<T extends Record<string, any>> {
```
Comment (matthewlipski) [inline packages/core/src/shared/EventEmitter.ts:13]: Is it necessary to have this file? What's the benefit vs the previous approach?

**Review by matthewlipski — COMMENTED** [review]
Nice! Much less code than before. As far as I can see it's mostly due to grouping the UI plugins in a single extension and and moving logic from `BlockNoteEditor` to `EventEmitter`. I am wondering though about using `EventEmitter` vs just adding similar logic to `BlockNoteEditor`, and I have a few questions about that.

**Pull request merged**